src/control/PlanarArmController.h

```diff
@@ -85,14 +86,18 @@ class PlanarArmController {
 	 * @return The new command, which is the new joint positions.
 	 */
 	void set_x_vel(robot::types::datatime_t currTime, double targetVel) {
+		Eigen::Vector2d pos;
 		std::lock_guard<std::mutex> lock(mutex);
+		pos = setpoint;
 		if (velTimestamp.has_value()) {
 			double dt = util::durationToSec(currTime - velTimestamp.value());
-			setpoint += velocity * dt;
+			pos += velocity * dt;
```

Comment:
If the `velTimestamp` doesn't have a value, there's no point in setting `setpoint` again, since we're not changing it. Can you change this to just `setpoint = normalizeEEWithinRadius(setpoint + velocity * dt)`?

After that you can remove the `pos` variable